src/main/resources/avro/common.avdl

```diff
@@ -387,19 +302,59 @@ record Position {
 }
 
 /**
-A `Segment` is a range on a sequence, possibly including the joins at the
-sequence's ends. It does not include base data. (The bases for a sequence are
-available through the `getSequenceBases()` API call.)
+A `Join` is simply a pair of Position objects, which we call the sides. The are
+logically unordered (i.e. swapping makes no difference), but we require a rank
+on the Sequences, and so implicitly on the positions, so to avoid ambiguity we
+require that the position for side1 is less than that for side2.
+*/
+record Join {
```

Comment:
I think we do; on balance I think not having a Sequence was a bad idea, even when it was possible to make Segment do all the work, so I think if it's a first-class object in the model it needs to have a type, if only to avoid confusing people.